### PR TITLE
IMAGEDAM-1666: Photosales icon is visible when syndication usage is present

### PR DESCRIPTION
**Pull request by Conalb97**
## What does this change?

**Note: this PR should be reviewed after #4267 as it builds on the changes included in that PR**

The relevant files for this PR are:
- kahuna/public/js/components/gr-icon/gr-icon.css
- kahuna/public/js/components/gr-icon/gr-icon.js
- kahuna/public/js/components/gr-icon/icons/sent-to-photosales.svg
- kahuna/public/js/preview/image.html
- kahuna/public/js/preview/image.js
- kahuna/public/js/services/image/usages.js

As an archivist user of BBC Images, after I have sent an image to BBC Photo Sales I want to see a visual indication that this has taken place. Therefore, after an image has been selected and sent to Photo Sales a 'sales' icon should appear in the thumbnail of that image.

The Photosales functionality is BBC specific, hence it is disabled by default via the showSendToPhotoSales feature flag.

## How should a reviewer test this change?

As a user with elevated permissions and with the showSendToPhotoSales flag set to true:

A reviewer can test this change by selecting an image and indicating that they want to send it to BBC Photo Sales. After this action has completed, a 'SALES' icon should appear in the thumnbail of that image, as demonstrated in the following screenshot:

Screenshot 2024-05-13 at 10 20 21

## How can success be measured?

This testing can be considered successful if after following the above steps, the 'SALES' icon appears on the selected image, as demonstrated in the previous screenshot. Additionally, images without a syndication usage should not have this icon present.  

## Tested? Documented?
- [x] locally by committer
- [x] locally by Guardian reviewer
- [ ] on the Guardian's TEST environment
- [ ] relevant documentation added or amended (if needed)

### CODE REVIEW

**Comment by paperboyo** [comment]
Just a note that we already do have icons associated with [different syndication statuses](https://github.com/guardian/grid/blob/main/common-lib/src/main/scala/com/gu/mediaservice/model/SyndicationStatus.scala#L5-L11):
- [review](https://media.gutools.co.uk/search?nonFree=true&syndicationStatus=review):
    image
- [queued](https://media.gutools.co.uk/search?nonFree=true&syndicationStatus=queued):
    image
- [blocked](https://media.gutools.co.uk/search?nonFree=true&syndicationStatus=blocked):
   image
- [sent](https://media.gutools.co.uk/search?nonFree=true&syndicationStatus=sent):
   image

Now, I have no idea how much of BBC syndication code and functionality overlaps Guardian’s. Or would it be helpful if it did. Just wanted to mention in case useful.

File: kahuna/public/js/preview/image.js [hunk 1]
```diff
@@ -64,6 +64,8 @@ image.controller('uiPreviewImageCtrl', [
           ctrl.imageAsArray = [newImage];
       });
 
+      ctrl.showSendToPhotoSales = () => $window._clientConfig.showSendToPhotoSales;
+
```
Comment (AndyKilmory) [inline kahuna/public/js/preview/image.js:68]: Does this need to be a function call here or can it be accessed directly as a variable? (changing to variable call would impact image.html)

**Review by rebecca-thompson — APPROVED** [review]
I have tested this locally and it all works as expected and the code looks good.

Just want to second @paperboyo's very valid comment that we already have an icon slot available for syndication status and it would be great to be able to configure that slot rather than having an increasing amount of Guardian-specific and BBC-specific code.

Perhaps a conversation for another time, but something we should definitely consider when migrating this codebase to React. 

IMO, making icons configurable would be a much more sustainable approach.

**Comment by Conalb97** [comment]
> I have tested this locally and it all works as expected and the code looks good.
> 
> Just want to second @paperboyo's very valid comment that we already have an icon slot available for syndication status and it would be great to be able to configure that slot rather than having an increasing amount of Guardian-specific and BBC-specific code.
> 
> Perhaps a conversation for another time, but something we should definitely consider when migrating this codebase to React.
> 
> IMO, making icons configurable would be a much more sustainable approach.

Thanks for the review Rebecca! Apologies for just getting round to responding to this now, I was on a training course earlier this week and this PR slipped my mind until now!

I agree that making icons configurable would be a good change and certainly something to look into for the future.

Would someone from the Guardian be able to merge this PR now it's approved? :)

**Pull request merged**